### ISSUE TITLE
Fix Arbitrary instances for color spaces / models with unusual coordinate ranges

### ISSUE THREAD

**Issue by gilgamec** [lehins/Color#6]
I noticed that the test specifications for e.g. `LAB` use the same `Abritrary` instance as others:
```haskell
instance (Elevator e, Random e, Illuminant i) => Arbitrary (Color (LAB (i :: k)) e) where
  arbitrary = ColorLAB <$> arbitraryElevator <*> arbitraryElevator <*> arbitraryElevator
```
where `arbitraryElevator` gives either the full range of values (for discrete quanta like `Word8`) or the range [0,1] for `Float` and `Double`. However, `LAB` at least (and possibly others) uses a different range: `L` can go from 0 to 100, while `a` and `b` vary in a similar range, depending on `L`, and can also be negative. Thus, right now the spec tests are only checking colors very close to black. I'm not sure how this can be handled within QuickCheck.

**Comment by lehins** [comment]
You are absolutely right about this. We need a new concept of min max values on per channel basis, which has to be at the `ColorModel` level in order to get this done properly. Arbitrary instance however can be fixed before we get it figure out a proper solution.
